### ISSUE TITLE
Server crashes from using Rainbow Chicken Spawn Egg

### ISSUE THREAD

**Issue by lunofe** [jayhillx/MysticsBiomes#15]
- Minecraft: 1.20.1
- Forge: 47.2.19
- MysticsBiomes: 1.20.1-3.1

Crashreport:
```
---- Minecraft Crash Report ----
// Daisy, daisy...

Time: 2024-01-03 20:24:47
Description: Exception in server tick loop

java.lang.NoClassDefFoundError: org/apache/commons/compress/utils/Lists
	at com.mysticsbiomes.common.entity.animal.RainbowChicken.getOffspringVariant(RainbowChicken.java:111) ~[mysticsbiomes-1.20.1-3.1.5.jar%23262!/:3.1] {re:classloading}
	at com.mysticsbiomes.common.entity.animal.RainbowChicken.getBreedOffspring(RainbowChicken.java:102) ~[mysticsbiomes-1.20.1-3.1.5.jar%23262!/:3.1] {re:classloading}
	at com.mysticsbiomes.common.entity.animal.RainbowChicken.m_142606_(RainbowChicken.java:40) ~[mysticsbiomes-1.20.1-3.1.5.jar%23262!/:3.1] {re:classloading}
	at net.minecraft.world.item.SpawnEggItem.m_43215_(SpawnEggItem.java:165) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.entity.Mob.m_21499_(Mob.java:1232) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:lithium.mixins.json:entity.inactive_navigations.MobEntityMixin,pl:mixin:APP:lithium.mixins.json:entity.skip_equipment_change_check.MobEntityMixin,pl:mixin:APP:scuba_gear.mixins.json:MobEntityMixin,pl:mixin:APP:vampirism.mixins.json:MixinMobEntity,pl:mixin:A}
	at net.minecraft.world.entity.Mob.m_6096_(Mob.java:1193) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:lithium.mixins.json:entity.inactive_navigations.MobEntityMixin,pl:mixin:APP:lithium.mixins.json:entity.skip_equipment_change_check.MobEntityMixin,pl:mixin:APP:scuba_gear.mixins.json:MobEntityMixin,pl:mixin:APP:vampirism.mixins.json:MixinMobEntity,pl:mixin:A}
	at net.minecraft.world.entity.player.Player.m_36157_(Player.java:1215) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:computing_frames,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_playerentity_iswearing,re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_playerentity_iswearing,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_playerentity_iswearing,pl:mixin:APP:vampiricageing.mixins.json:PlayerMixin,pl:mixin:APP:werewolves.mixins.json:entity.PlayerEntityMixin,pl:mixin:APP:vampirism.mixins.json:MixinPlayerEntity,pl:mixin:A,re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_playerentity_iswearing}
	at net.minecraft.server.network.ServerGamePacketListenerImpl$2.performInteraction(ServerGamePacketListenerImpl.java:2170) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading}
	at net.minecraft.server.network.ServerGamePacketListenerImpl$2.m_142299_(ServerGamePacketListenerImpl.java:2187) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading}
	at net.minecraft.network.protocol.game.ServerboundInteractPacket$InteractionAction.m_142457_(ServerboundInteractPacket.java:124) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading}
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.m_179617_(ServerboundInteractPacket.java:75) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.network.ServerGamePacketListenerImpl.m_6946_(ServerGamePacketListenerImpl.java:2133) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:badpackets.mixins.json:MixinServerGamePacketListenerImpl,pl:mixin:APP:craterlib.mixins.json:events.ServerGamePacketListenerImplMixin,pl:mixin:APP:mixins/common/nochatreports.mixins.json:server.MixinServerGamePacketListenerImpl,pl:mixin:A,re:mixin,pl:accesstransformer:B}
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.m_5797_(ServerboundInteractPacket.java:62) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.network.protocol.game.ServerboundInteractPacket.m_5797_(ServerboundInteractPacket.java:12) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.network.protocol.PacketUtils.m_263899_(PacketUtils.java:25) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading}
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:classloading}
	at net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.m_6367_(ReentrantBlockableEventLoop.java:23) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:computing_frames,re:mixin,re:classloading}
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:1029) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_6367_(MinecraftServer.java:173) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:1012) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:1006) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B}
	at net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:992) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:908) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:281) ~[server-1.20.1-20230612.114412-srg.jar%23298!/:?] {re:mixin,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,re:classloading,pl:accesstransformer:B,xf:fml:xaeroworldmap:xaero_wm_minecraftserver,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:1589) [?:?] {re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin}
Caused by: java.lang.ClassNotFoundException: org.apache.commons.compress.utils.Lists
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:157) ~[securejarhandler-mohist-2.1.11.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?] {re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin,re:mixin}
	... 28 more
```

**Comment by jayhillx** [comment]
oh geez sorry you’re experiencing that. i’ll check it out and have it updated this weekend!

**Comment by lunofe** [comment]
No worries, it should only occur when using a spawn egg, so it's not really an issue a regular player would encounter 🙂

**Comment by jayhillx** [comment]
this is now fixed and will be in update 3.1.6! :)
